src/tiled/mainwindow.cpp

```diff
@@ -1129,6 +1131,33 @@ void MainWindow::setFullScreen(bool fullScreen)
         setWindowState(windowState() & ~Qt::WindowFullScreen);
 }
 
+void MainWindow::toggleClearView(bool clearView)
+{
+    QList<QDockWidget*> docks = this->findChildren<QDockWidget*>();
+    QList<QToolBar*> toolbars = this->findChildren<QToolBar*>();
+
+    if (clearView) {
+        mHiddenDocks.clear();
+        mHiddenToolbars.clear();
+
+        for (auto dock : docks) {
+            if (dock->isVisible())
+                mHiddenDocks.push_back(dock);
+            dock->hide();
+        }
+        for (auto toolbar : toolbars) {
+            if (toolbar->isVisible())
+                mHiddenToolbars.push_back(toolbar);
+            toolbar->hide();
+        }      
+    } else {
+        for (auto dock : mHiddenDocks)
+            dock->show();
+        for (auto toolbar : mHiddenToolbars)
+            toolbar->show();
```

Comment:
Please clear the containers after this as well, to make sure that repeated calls of `toggleClearView(false)` (like may happen during shutdown) don't make in-the-meantime-manually-hidden docks visible again.